### PR TITLE
[BugFix] fix npe for rule-based optimizer timeout

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

fix npe for rule-based optimizer timeout

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java [hunk 1]
```diff
@@ -42,7 +43,9 @@ public void executeTasks(TaskContext context) {
             if (timeout > 0 && watch.elapsed(TimeUnit.MILLISECONDS) > timeout) {
                 // Should have at least one valid plan
                 // group will be null when in rewrite phase
-                Group group = context.getOptimizerContext().getMemo().getRootGroup();
+                // memo may be null for rule-based optimizer
+                Memo memo = context.getOptimizerContext().getMemo();
+                Group group = memo == null ? null : memo.getRootGroup();
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java:48]: should check why timeout?

**Review by Seaven — DISMISSED** [review]
Why timeout?

**Review by murphyatwork — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java|2|2|100.00%|[]|

**Review by kangkaisen — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by ABingHuang** [comment]
run starrocks_admit_test

**Comment by ABingHuang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java|2|2|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**